test/support/middleware_example.rb

```diff
@@ -99,7 +100,7 @@ def run
       end
 
       def filter_sensitive_data
-        output[:spell] = '***'
+        output[:spell] = '***'.dup
```

Comment:
Should I want to know why? xD

Reply:
We do some modifications of these strings using `<<`. By adding the magic comment (`# frozen_string_literal: true`) all string literals become frozen by default and exceptions are raised when we try to modify them. `dup`ing a frozen string creates an un-frozen copy of it, which can be modified.